src/core/border_router/infra_if_platform.cpp

```diff
@@ -52,4 +52,16 @@ extern "C" void otPlatInfraIfRecvIcmp6Nd(otInstance *        aInstance,
     instance.Get<BorderRouter::RoutingManager>().RecvIcmp6Message(
         aInfraIfIndex, static_cast<const Ip6::Address &>(*aSrcAddress), aBuffer, aBufferLength);
 }
+
+extern "C" void otPlatInfraIfStateChanged(otInstance *        aInstance,
```

Comment:
Why is this method a `otPlatXxxx`? I think this is not a **platform** method that is to be defined by each platform, but a public OT API method that is defined in OT core and used by each platform. 

```suggestion
void otInfraIfNotifyStateChanged(otInstance *        aInstance,
```

Reply:
Methods definitions in `platform/` are meant to be used/called by platform modules but not to be implemented by them. Examples can be found for `radio` and `uart`:
https://github.com/openthread/openthread/blob/14ef4e4082ca06cdd05eb916cf8414b40c88c230/include/openthread/platform/radio.h#L723-L751

https://github.com/openthread/openthread/blob/14ef4e4082ca06cdd05eb916cf8414b40c88c230/include/openthread/platform/uart.h#L99-L112